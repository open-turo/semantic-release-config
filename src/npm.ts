import config from "./";

// We have a set of plugins that ideally should run after every other plugin to guarantee that things like publishing to NPM
// already happen before these plugins run
const pluginsThatGoAtTheEnd = new Set([
  "@semantic-release/exec",
  "semantic-release-fotingo",
]);

/**
 * Given a semantic release plugin config, return the plugin name
 * @param p Plugin config
 * @returns Plugin name
 */
const getPluginName = (p: typeof config.plugins[number]) => {
  if (typeof p === "string") {
    return p;
  }
  return p[0];
};

/**
 * Semantic release configuration for NPM projects. It extends the default configuration by publishing to NPM
 * and creating a Pull Request with version updates in files like package.json
 */
export = {
  plugins: [
    ...config.plugins.filter(
      (p) => !pluginsThatGoAtTheEnd.has(getPluginName(p))
    ),
    [
      "@semantic-release/npm",
      {
        tarballDir: "pack",
      },
    ],
    [
      "@semantic-release/git",
      {
        assets: ["package.json", "package-lock.json", "README.md", "yarn.lock"],
        message:
          "ci(release): ${nextRelease.version} <% nextRelease.channel !== 'next' ? print('[skip ci]') : print('') %>\n\n${nextRelease.notes}",
      },
    ],
    ...config.plugins.filter((p) =>
      pluginsThatGoAtTheEnd.has(getPluginName(p))
    ),
  ],
};
