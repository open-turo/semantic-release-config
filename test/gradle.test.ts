import npm from "~/gradle";

describe("gradle", () => {
  test("adds gradle-semantic-release-plugin and semantic-release/git plugins", () => {
    const gradlePlugin = npm.plugins[3];
    const semanticReleasePlugin = npm.plugins[4];
    expect(gradlePlugin).toMatchSnapshot();
    expect(semanticReleasePlugin).toMatchSnapshot();
  });
});
