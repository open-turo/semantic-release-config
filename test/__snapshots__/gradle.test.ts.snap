// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gradle adds gradle-semantic-release-plugin and semantic-release/git plugins 1`] = `"gradle-semantic-release-plugin"`;

exports[`gradle adds gradle-semantic-release-plugin and semantic-release/git plugins 2`] = `
[
  "@semantic-release/git",
  {
    "assets": [
      "gradle.properties",
      "README.md",
    ],
    "message": "ci(release): \${nextRelease.version} <% nextRelease.channel !== 'next' ? print('[skip ci]') : print('') %>

\${nextRelease.notes}",
  },
]
`;
